hera_cal/delay_filter.py

```diff
@@ -166,7 +200,75 @@ def partial_load_delay_filter_and_write(infilename, calfile=None, Nbls=1,
                                filled_outfilename=filled_outfilename, partial_write=True,
                                clobber=clobber, add_to_history=add_to_history)
         df.hd.data_array = None  # this forces a reload in the next loop
-        
+
+
+def partial_load_dayenu_delay_filter_and_write(infilename, calfile=None, Nbls=1, spw_range=None, cache_dir=None,
+                                               res_outfilename=None, CLEAN_outfilename=None, filled_outfilename=None,
+                                               clobber=False, add_to_history='', write_cache=False, read_cache=True,
+                                               **filter_kwargs):
+    '''
+    Uses partial data loading and writing to perform delay filtering.
+
+    Arguments:
+        infilename: string path to data to uvh5 file to load
+        cal: optional string path to calibration file to apply to data before delay filtering
+        Nbls: the number of baselines to load at once.
+        spw_range: spw_range of data to delay-filter.
+        cache_dir: string, optional, path to cache file that contains pre-computed dayenu matrices.
+                    see uvtools.dspec.dayenu_filter for key formats.
+        res_outfilename: path for writing the filtered visibilities with flags
+        clobber: if True, overwrites existing file at the outfilename
+        add_to_history: string appended to the history of the output file
+        write_cache: bool. If true, create new cache file with precomputed matrices
+                           that were not in previously loaded cache files.
+        read_cache: bool, If true, read existing cache files in cache_dir before running.
+        filter_kwargs: additional keyword arguments to be passed to DelayFilter.run_dayenu_foreground_filter()
+    '''
+    # Load up the cache file with the most keys (precomputed filter matrices).
+    cache = {}
+    if cache_dir is not None and read_cache:
```

Comment:
you probably want to pull out this cache io and put it in io or even in uvtools. Looks like you use it a lot.

Reply:
i've put it in io.